src/werkzeug/routing.py

```diff
@@ -1650,9 +1650,13 @@ def bind_to_environ(
         environ = _get_environ(environ)
         wsgi_server_name = get_host(environ).lower()
         scheme = environ["wsgi.url_scheme"]
+        connection = [
```

Comment:
I was thinking of suggesting `parse_set_header`, but technically none of the values in the Connection header should need to be quoted, so that would be a bunch of parsing and processing overhead for no reason. I'll just leave it here, but doesn't need to change.

```python
from .http import parse_set_header

connection = parse_set_header(environ.get("HTTP_CONNECTION"))

if "upgrade" in connection ...
```

Reply:
The `parse_list_header` function should work better than `parse_set_header`, as it returns a plain list with the tokens. But it is definitely doing extra work by checking for quoted tokens, which as far as I know are never used. I would be fine switching to `parse_list_header` and simplify this logic, let me know if you'd prefer it done that way.

Reply:
The set header deals with case-insensitive matching as well. Just leave it as-is, I think we're fine with simple string parsing if we don't need the more complex features.